### PR TITLE
[codex] Fuse Gemma CUDA greedy decode tail

### PR DESCRIPTION
**Pull request by DeanoC**
## Summary

- Fuse Gemma CUDA BF16 batch-1 greedy decode input staging, persistent decode, final RMSNorm, LM-head scan, and argmax into one CUDA launch when the per-layer embedding table is resident.
- Keep the existing standalone final RMSNorm + LM-head argmax path as a fallback.
- Rework the CUDA greedy tail so every warp in a block scans a distinct vocab row stream, instead of redundantly scanning the same rows and discarding seven warp results.

## Impact

This keeps validation/full-logits paths unchanged while improving the non-validation greedy decode path for Gemma CUDA E2B BF16.

Perf on sm86 for `gemma4-e2b`, prompt `Hello, world`, 64 generated tokens:

- Before tail row-distribution fix in this pass: `decode_ms=833`, `ms_per_step=13`
- After: `decode_ms=563`, `ms_per_step=9`

The generated token stream was unchanged.

## Validation

- `SUPERSONIC_BACKENDS=cuda cargo build --release --bin supersonic`
- `HF_HOME=/dev/shm/hf_home_gemma4 SUPERSONIC_BACKENDS=cuda TIMEOUT=900 ./tests/sm86/run_gemma4.sh /dev/shm/gemma-4-E2B`
  - `token_mismatches=0`
  - `decode_max_delta=0.3542`
- Greedy timing:
  - `HF_HOME=/dev/shm/hf_home_gemma4 SUPERSONIC_BACKENDS=cuda timeout 900 ./target/release/supersonic --backend cuda --model gemma4-e2b --model-dir /dev/shm/gemma-4-E2B --prompt "Hello, world" --max-new-tokens 64 --batch-size 1`
  - `decode_ms=563`, `ms_per_step=9`

The downloaded model/cache were cleaned from `/dev/shm` after validation.

### CODE REVIEW

**Comment by DeanoC** [comment]
@codex review
